### PR TITLE
change some state to readonly

### PR DESCRIPTION
**Pull request by chenshuai2144**
look https://github.com/DefinitelyTyped/DefinitelyTyped/pull/26813

Although not agreeing, it is still compatible with it

### CODE REVIEW

File: components/checkbox/Group.tsx [hunk 1]
```diff
@@ -54,13 +54,9 @@ export default class CheckboxGroup extends React.Component<CheckboxGroupProps, C
...
   };
-
-  constructor(props: CheckboxGroupProps) {
-    super(props);
-    this.state = {
-      value: props.value || props.defaultValue || [],
-     };
-  }
+  readonly state = {
+    value: this.props.value || this.props.defaultValue || [],
```
Comment (afc163) [inline components/checkbox/Group.tsx]: 这样不会报错么，能读到 this.props ？

Comment (chenshuai2144) [reply components/checkbox/Group.tsx]: 是可以的  typescript 会帮助我们自动的 constructor。
```
constructor() {
        super(...arguments);
        this.state = {
            value: this.props.value || this.props.defaultValue || [],
        };
       .........
    }
```

Comment (afc163) [reply components/checkbox/Group.tsx]: 在 constructor 里用 this.props 可能会有坑。http://imweb.io/topic/59dc5a8b856028aa249e2a58

Comment (afc163) [reply components/checkbox/Group.tsx]: 要验证一下：

https://github.com/ant-design/ant-design/issues/1492
http://leonshi.com/2016/03/16/babel6-es6-inherit-issue-in-ie10/
https://github.com/babel/babel/issues/3041

**Comment by afc163** [comment]
貌似不需要改了，@types/react@16.4.6 修复了。

**Comment by chenshuai2144** [comment]
过分

**Pull request closed without merging**